File: public/usage-examples/input/mouse_position-1-example.py
from splashkit import *

open_window("Mouse Position Display", 800, 600)

while not quit_requested():
    process_events()

    mouse_point = mouse_position()

    clear_screen(color_white())

    draw_text_no_font_no_size("Move the mouse to track its position in the window.", color_black(), 20, 20)
    draw_text_no_font_no_size("Mouse X: " + str(mouse_point.x), color_black(), 20, 60)
    draw_text_no_font_no_size("Mouse Y: " + str(mouse_point.y), color_black(), 20, 100)

    fill_circle(color_blue(), mouse_point.x, mouse_point.y, 8)
    draw_circle(color_black(), mouse_point.x, mouse_point.y, 8)

    refresh_screen(60)

close_all_windows()

File: public/usage-examples/input/mouse_position-1-example.cpp
#include "splashkit.h"

int main()
{
    open_window("Mouse Position Display", 800, 600);

    while (!quit_requested())
    {
        process_events();

        point_2d mouse_point = mouse_position();

        clear_screen(COLOR_WHITE);

        draw_text("Move the mouse to track its position in the window.", COLOR_BLACK, 20, 20);
        draw_text("Mouse X: " + to_string(mouse_point.x), COLOR_BLACK, 20, 60);
        draw_text("Mouse Y: " + to_string(mouse_point.y), COLOR_BLACK, 20, 100);

        fill_circle(COLOR_BLUE, mouse_point.x, mouse_point.y, 8);
        draw_circle(COLOR_BLACK, mouse_point.x, mouse_point.y, 8);

        refresh_screen(60);
    }

    close_all_windows();

    return 0;
}